### PR TITLE
[Swift] Add support for BridgeObject-based Set/Dictionary

### PR DESCRIPTION
**Pull request by lorentey**
In https://github.com/apple/swift/pull/19688, we’re switching the variant type of Set and Dictionary from a custom enum to a BridgeObject. Add support for this to lldb.

### CODE REVIEW

**Comment by lorentey** [comment]
@swift-ci please test

**Comment by lorentey** [comment]
This adds yet another variant to the pile of supported layouts, with some parts going back to Swift 3 or even earlier. 

I'll submit a cleanup removing obsolete variants after this lands.

File: source/Plugins/Language/Swift/SwiftHashedContainer.cpp [hunk 1]
```diff
@@ -910,19 +910,22 @@ HashedStorageHandlerUP
 HashedCollectionConfig::CreateHandler(ValueObject &valobj) const {
   static ConstString g__variant("_variant"); // Swift 5
   static ConstString g__variantBuffer("_variantBuffer"); // Swift 4
-  static ConstString g_native("native");
-  static ConstString g_cocoa("cocoa");
+  static ConstString g_native("native"); // Swift 4.2
```
Comment (dcci) [inline source/Plugins/Language/Swift/SwiftHashedContainer.cpp]: Feel free to remove those. (and the `_variantBuffer` above)

**Comment by lorentey** [comment]
Please kindly test with the following PR:
https://github.com/apple/swift/pull/19688
@swift-ci test

**Comment by lorentey** [comment]
> 21:01:54 AssertionError: False is not True : 'frame variable -d run -- d2' returns expected result, got '([NSObject : AnyObject]) d2 = {}'
> ..
> 21:03:44 FAIL: test_dictionary_nsobject_any_object_dsym (lang/swift/variables/dict_nsobj_anyobj/TestSwiftDictionaryNSObjectAnyObject.py)
> 21:03:44 FAIL: test_dictionary_nsobject_any_object_dwarf (lang/swift/variables/dict_nsobj_anyobj/TestSwiftDictionaryNSObjectAnyObject.py)

Hm, the tests might not have picked up the correct swiftc.

**Comment by lorentey** [comment]
https://github.com/apple/swift/pull/19688
@swift-ci test

**Comment by lorentey** [comment]
🤔 Looks like some of the tests might be using a previous compiler release, or that we still depend on some aspect of the obsolete layouts. I’ll spin the removals off to a separate PR.

**Comment by lorentey** [comment]
Or is it actually a bug in the new layout formatter?

@swift-ci test

**Comment by lorentey** [comment]
apple/swift#19688
@swift-ci test

**Pull request merged**